### ISSUE TITLE
Prevent snapshot rollback if the live dataset is in use

### ISSUE THREAD

**Issue by shodanshok** [openzfs/zfs#11482]
### Describe the feature would like to see added to OpenZFS
When operating on a live dataset, ZFS behaves inconsistently:
 - `zfs destroy tank/live` will fail with a "target busy" error
 - `zfs rollback tank/live@oldsnap` will *succeed* with not error.

The issue is that, when operating on a live dataset, a `rollback` can wreak havok on the integrity of the application running on the live dataset. As an example, think about a busy virtual machine or database whose data is suddenly and unknowingly rolled-back: such an operation will cause immediate (and often irreversible) data corruption.

We should avoid `zfs rollback` to succeed when the live dataset is busy. If deemed really necessary, the "force" flag (`-f`) can be used to coerce the system to rollback the target.

### How will this feature improve OpenZFS?
It will avoid accidental, but serious, application data corruption.

### Additional context
None.

**Comment by DeHackEd** [comment]
I have an objection. Yes the app might break when the files change under them. However "irreversible" corruption I disagree with. In fact it's fairly easily corrected - kill the app, do a rollback again to clean up the corruption, and restart it.

**Comment by shodanshok** [comment]
@DeHackEd Yeah, but any write done by the app/vm after latest snapshot will be lost. As the detection of such corruption can take quite some time (especially when running virtual machines), we risk losing much valuable data. As ZFS is all about data integrity/safety, it seems quite reasonable to be to ask for a force flag before doing anything which can led to data loss.

**Comment by PaulZ-98** [comment]
@shodanshok in the `zfs destroy` case the EBUSY comes from unmount that zfs issues before the destroy.  If your rollback is scripted (i.e. not randomly typed in by a user), you could consider calling unmount before rollback, and testing for EBUSY before letting the rollback occur.
